### ISSUE TITLE
Reader: improve accessibility of post card opening

### ISSUE THREAD

**Issue by bluefuton** [Automattic/wp-calypso#24747]
Reader post cards currently use a click handler attached to a `li` or `div` to open the full post. 

There's no keyboard handler on the full card (but there keyboard functionality on the post title). We also shouldn't be attaching click events to non-interactive elements.

https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-to-interactive-role.md

Here's the eslint output:

```
wp-calypso/client/blocks/reader-combined-card/post.jsx
  124:4  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  124:4  error  Non-interactive elements should not be assigned mouse or keyboard event listeners               jsx-a11y/no-noninteractive-element-interactions

wp-calypso/client/blocks/reader-post-card/compact.jsx
  20:3  error  Visible, non-interactive elements with click handlers must have at least one keyboard listener  jsx-a11y/click-events-have-key-events
  20:3  error  Non-interactive elements should not be assigned mouse or keyboard event listeners               jsx-a11y/no-noninteractive-element-interactions
```

Discovered while working on lint fixes in #24702.

**Comment by bluefuton** [comment]
Some good markup examples with `article` here: https://www.w3.org/WAI/GL/wiki/Using_HTML5_article_element

**Comment by bluefuton** [comment]
A related issue here: https://github.com/twbs/bootstrap/issues/18294

**Comment by ryelle** [comment]
Inclusive Components just published a write up on [an accessible card pattern](https://inclusive-components.design/cards/), which might be helpful (our cards might also be too complex for their examples)

**Comment by roo2** [comment]
I don't think it's neccessary to make any changes here for accessibility. The first principle is that people should be able to use the site with a screenreader, not that the same html elements MUST be used by a screenreader as by the mouse. e.g. sometimes special content is only available to screenreaders https://webaim.org/techniques/css/invisiblecontent/.  In our case, we have buttons available that screenreader users can tab through and click to view the post details.

If we try to make the whole card tab-able and actionable via keyboard, that will interfere with the buttons inside the card. We're not supposed to nest clickable elements. If the card is a "button" accessibility wise, it MUST not have descendants that also act as buttons https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

I think the best is just to leave this as is. If we try to make the whole post/card clickable we will just make accessibility worse.

**Comment by roo2** [comment]
@bluefuton I couldn't work out what lint command returns that error, when I ran `yarn eslint:js` it doesn't return errors or warnings for those files (in 2023, maybe something has changed since this was reported)
